.github/workflows/build_and_test.yml

```diff
@@ -71,17 +73,37 @@ jobs:
              export IPDK_RECIPE=`pwd`
              cd $IPDK_RECIPE/setup
              cmake -B build -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/deps
-             cmake --build build -j6
+             cmake --build build -j6   
 
+      - name: 'Run Stratum unit tests'
+        run: |
+             cd /tmp
+             curl -o bazel-4.2.3-installer-linux-x86_64.sh https://github.com/bazelbuild/bazel/releases/download/4.2.3/bazel-4.2.3-installer-linux-x86_64.sh
+             chmod +x bazel-4.2.3-installer-linux-x86_64.sh
+             ./bazel-4.2.3-installer-linux-x86_64.sh --user
+             export PATH="$PATH:$HOME/bin"
+             cd ${GITHUB_WORKSPACE}/ipdk.recipe/stratum/stratum
+             export DPDK_INSTALL=${GITHUB_WORKSPACE}/install
+             echo "DPDK" > ${GITHUB_WORKSPACE}/install/share/TARGET
+             xargs -a ${GITHUB_WORKSPACE}/ipdk.recipe/.github/dpdk-tests.txt bazel test --define target=dpdk
+             cd bazel-testlogs/stratum
+             zip stratum_ut_logs.zip `find . -name test.xml -print`
+                          
+      - name: 'Upload stratum-ut-result artifact'
+        uses: actions/upload-artifact@v3
+        with:
+          name: stratum-ut-result
+          path: ipdk.recipe/stratum/stratum/bazel-testlogs/stratum/stratum_ut_logs.zip
+          
       - name: 'Execute dpdk target build'
         run: |
              cd ${GITHUB_WORKSPACE}/ipdk.recipe
              export IPDK_RECIPE=${GITHUB_WORKSPACE}/ipdk.recipe
              export SDE_INSTALL=${GITHUB_WORKSPACE}/install
              export DEPEND_INSTALL=${GITHUB_WORKSPACE}/deps
              source ./scripts/dpdk/setup_env.sh $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL
-             ./make-all.sh --target=dpdk
-
+             ./make-all.sh --target=dpdk  --prefix=${GITHUB_WORKSPACE}/deps 
```

Comment:
For example:
```
INSTALL
├── deps
│   ├── bin
│   ├── include
│   ├── lib
│   └── share
├── dpdk-target
│   ├── bin
│   ├── include
│   ├── lib
│   ├── sbin
│   └── share
├── es2k-target
│   ├── bin
│   ├── include
│   ├── lib
│   ├── sbin
│   └── share
└── ovs
    ├── bin
    ├── etc
    ├── include
    ├── lib
    ├── sbin
    ├── share
    └── var
```

Reply:
How are we going to get es2k-sde, as it would be developed in innersource model?

Reply:
Thanks Derek for the suggestions!
can i consider the suggestions in next PR?

Reply:
> How are we going to get es2k-sde, as it would be developed in innersource model?

The SDE will need to be prebuilt from innersource, packaged, and somehow made available to CI.

In what context are GitHub actions run? Is there a special user account associated with the CI workflows? If so, perhaps we could create a private repository on ipdk-io, archive versions of the binary package there, and grant the CI account read access to the repository. The only gotcha I can think is providing a mechanism to specify the version of the binary package the workflow should fetch.

Come to think of it, we might need to test periodically against different versions of the SDE to verify backward compatibility.

Yikes. The problem space gets larger and more complicated every time I think about it.